include/dlaf/factorization/cholesky/mc/cholesky_L.h

```diff
@@ -87,191 +197,61 @@ void cholesky_L(Matrix<T, Device::CPU>& mat_a) {
 // Distributed implementation of Lower Cholesky factorization.
 template <class T>
 void cholesky_L(comm::CommunicatorGrid grid, Matrix<T, Device::CPU>& mat_a) {
-  using hpx::util::unwrapping;
   using hpx::threads::executors::pool_executor;
   using hpx::threads::thread_priority_high;
   using hpx::threads::thread_priority_default;
   using comm::internal::mpi_pool_exists;
-  using common::internal::vector;
+  using ConstTile_t = matrix::Tile<const T, Device::CPU>;
 
-  using ConstTile_t = typename Matrix<T, Device::CPU>::ConstTileType;
-  using Tile_t = typename Matrix<T, Device::CPU>::TileType;
-  using MemView_t = memory::MemoryView<T, Device::CPU>;
-
-  constexpr auto NonUnit = blas::Diag::NonUnit;
-  constexpr auto ConjTrans = blas::Op::ConjTrans;
-  constexpr auto NoTrans = blas::Op::NoTrans;
-  constexpr auto Right = blas::Side::Right;
-  constexpr auto Lower = blas::Uplo::Lower;
-
-  // Set up executor on the default queue with high priority.
   pool_executor executor_hp("default", thread_priority_high);
-  // Set up executor on the default queue with default priority.
   pool_executor executor_normal("default", thread_priority_default);
+
   // Set up MPI executor
+  comm::Index2D this_rank = grid.rank();
   auto executor_mpi = (mpi_pool_exists()) ? pool_executor("mpi", thread_priority_high) : executor_hp;
+  common::Pipeline<comm::CommunicatorGrid> mpi_task_chain(std::move(grid));
 
-  auto col_comm_size = grid.colCommunicator().size();
-  auto row_comm_size = grid.rowCommunicator().size();
-
-  const matrix::Distribution& distr = mat_a.distribution();
-
+  matrix::Distribution const& distr = mat_a.distribution();
   SizeType nrtile = mat_a.nrTiles().cols();
 
-  auto localnrtile_rows = distr.localNrTiles().rows();
-  auto localnrtile_cols = distr.localNrTiles().cols();
-
-  common::Pipeline<comm::CommunicatorGrid> serial_comm(std::move(grid));
-
-  auto this_rank = grid.rank();
-
   for (SizeType k = 0; k < nrtile; ++k) {
     // Create a placeholder that will store the shared futures representing the panel
-    vector<hpx::shared_future<ConstTile_t>> panel(distr.localNrTiles().rows());
-
-    auto kk_idx = GlobalTileIndex(k, k);
-    auto kk_tile_rank = distr.rankGlobalTile(kk_idx);
-
-    if (this_rank.col() == kk_tile_rank.col()) {
-      hpx::shared_future<ConstTile_t> kk_tile;
-
-      if (this_rank.row() == kk_tile_rank.row()) {
-        LocalTileIndex kk = distr.localTileIndex(kk_idx);
+    std::unordered_map<SizeType, hpx::shared_future<ConstTile_t>> panel;
```

Comment:
What is the benefit of using `std::unordered_map` over `std::vector`?

Reply:
I have to use `std::unordered_map` because I am using global indices which may have large gaps in-between as keys.

Reply:
ah yeah, good point! I didn't thought about it. because you changed everything to global indices, I forgot that.